core/src/main/java/com/exadel/aem/toolkit/bundle/customlists/datasource/ChildResourcesDatasource.java

```diff
@@ -139,43 +145,47 @@ public String getResourceType() {
      * Retrieves the list of item resources, which are either custom lists themselves, or folders
      * that may contain lists inside
      *
-     * @param resolver an instance of ResourceResolver
-     * @param children {@code Iterator<Resource>} instance used as the source of markup
+     * @param resolver An instance of ResourceResolver
+     * @param parent {@code Resource} instance used as the source of markup
      * @return a list of {@link Resource}s
      */
-    private List<Resource> getValidChildren(ResourceResolver resolver, Iterator<Resource> children) {
-        List<Resource> resources = new ArrayList<>();
-        while (children.hasNext()) {
-            Resource child = children.next();
-            Resource childParameters = resolver.getResource(child.getPath() + PATH_TO_JCR_CONTENT);
-            if (childParameters != null) {
-                String template = childParameters.getValueMap().get(NN_TEMPLATE, String.class);
-                if (template != null && template.equals(TEMPLATE)) {
-                    resources.add(child);
-                }
-            }
-            resources.addAll(getValidChildren(child));
-        }
-        return resources;
+    private List<Resource> getValidChildren(ResourceResolver resolver, Resource parent) {
+        return getChildrenStream(parent)
+            .filter(resource -> checkResourceAsTemplate(resolver, resource) || checkResourceAsFolder(resource))
+            .collect(Collectors.toList());
     }
 
     /**
-     * Retrieves the list of item resources, which are either pages, or folders that may contain lists inside
+     * Checks whether the resource is a custom list
```

Comment:
Please note that such kind of description may need to be altered throughout the project after we make ultimate decision on the "brand name" of lists feature. "Custom lists" are no good because they imply there are "non-custom", or "regular" lists which is not tot he fact. ACS AEM Common lists are as much "custom" as ours because not a part of the core

Reply:
"Custom Lists" will be renamed to "AAT Lists" per our agreement a little bit later, when the main part of the PR is handled, but before merging